sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/executeImmediate.scala

```diff
@@ -122,11 +122,14 @@ class SubstituteExecuteImmediate(val catalogManager: CatalogManager)
               resolveArguments(expressions))
           } else {
             val aliases = expressions.collect {
-              case (e: Alias) => e
+              case e: Alias => e
+              case u: UnresolvedAttribute => Alias(u, u.name)()
```

Comment:
seems we can just collect `NamedExpression`.

Reply:
can you be more specified with a real example?

Reply:
If we use
```
val namedExpressions = expressions.collect {
    case e: NamedExpression => e
}
```
for the following query
```
EXECUTE IMMEDIATE 'SELECT :parm' USING parm;
```
we get the following column name
`variablereference(system.session.parm='Hello')` instead of just `parm`.
i.e. the output of `query.show()` prints:
```
+----------------------------------------------+
|variablereference(system.session.parm='Hello')|
+----------------------------------------------+
|                                         Hello|
+----------------------------------------------+
```

Reply:
After digging deeper, it turns out we do not need to filter literals, there is a match for that:
`resolveArguments`

Reply:
got it, thanks!

Reply:
I can do the following in `sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/VariableReference.scala`:
```
  override def toString: String = {
    // REMOVE >> val qualifiedName = (catalog.name +: identifier.namespace :+ identifier.name).map(quoteIfNeeded)
    // REMOVE >> s"$prettyName(${qualifiedName.mkString(".")}=${varDef.currentValue.sql})"
    identifier.name // << ADD
  }
```

This allows for 

> seems we can just collect `NamedExpression`.

> ```
> val namedExpressions = expressions.collect {
>     case e: NamedExpression => e
> }
> ```
> 
> for the following query
> 
> ```
> EXECUTE IMMEDIATE 'SELECT :parm' USING parm;
> ```

The output is as follows: 
```
+-----+
|parm |
+-----+
|Hello|
+-----+
```
which is the correct output.

This is @srielau 's code which I do not know if I should edit (although tests pass with this mod).

@srielau @cloud-fan if this should be edited let me know, if not the Alias fix is ready to go.